app/Geocoding/GeocodingServiceProvider.php

```diff
@@ -42,7 +44,8 @@ function () use ($container) {
 
                 return new CachedGeocodingService(
                     $geocodingService,
-                    $container->get('cache')('geocoords')
+                    $container->get('cache')('geocoords'),
+                    empty($container->get('config')['address_enrichment']) ? new CoordinateEncoder() : new RichEncoder()
```

Comment:
While a featureflag is a good idea, I would avoid changing the existing code/functionality because that could introduce bugs both with and without the featureflag.